docs/src/main/asciidoc/dynamodb-guide.adoc

```diff
@@ -0,0 +1,751 @@
+
+DynamoDB is a scalable AWS managed *NoSQL* database.
+It supports both key-value and document data models, that enables to have a flexible schema for your data.
+This extension provides functionality that allows the client to communicate with the service when runninng in Quarkus.
+More information about DynamoDB you can find at https://aws.amazon.com/dynamodb/[Amazon DynamoDB].
+
+NOTE: The DynamoDB extension is based on https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/welcome.html[AWS Java SDK 2.x].
+It's major rewrite of the 1.x code base that offers two programming models (Blocking & Async).
+Keep in mind it's actively developed and does not support yet all the features available in SDK 1.x such as https://github.com/aws/aws-sdk-java-v2/issues/36[Document APIs], https://github.com/aws/aws-sdk-java-v2/issues/35[Object Mappers]
+
+The Quarkus extension supports two programming models:
+
+* Blocking access using Apache Http Client
+* https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/basics-async.html[Asynchronous programming] based on JDK's `CompletableFuture` objects and Netty Http client.
+
+In this guide, we see how you can get your REST services to use the DynamoDB locally and on AWS account.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* an IDE
+* Apache Maven 3.5.3+
+* AWS Account to access DynamoDB service
+* Optionally, Docker for your system to run dynamoDB locally for testing purposes
+
+=== Setup DynamoDB locally
+
+The easiest way to start work with DynamoDB is to run local instance as container.
+
+[source,shell,subs="verbatim,attributes"]
+----
+docker run --publish 8000:8000 amazon/dynamodb-local:1.11.477 -jar DynamoDBLocal.jar -inMemory -sharedDb
+----
+
+This starts a DynamoDB instance, that's accessible on port `8000`.
+You can check it running accessing the web shell on `http://localhost:8000/shell`.
+
+Have a look at the https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html[Setting Up DynamoDB Local] for other options to run DynamoDB.
+
+Open `http://localhost:8000/shell` in your browser.
+
+Copy & paste a following code to the shell and run it.
+[source,js,subs="verbatim,attributes"]
+----
+var params = {
+    TableName: 'QuarkusFruits',
+    KeySchema: [{ AttributeName: 'fruitName', KeyType: 'HASH' }],
+    AttributeDefinitions: [{  AttributeName: 'fruitName', AttributeType: 'S', }],
+    ProvisionedThroughput: { ReadCapacityUnits: 1, WriteCapacityUnits: 1, }
+};
+
+dynamodb.createTable(params, function(err, data) {
+    if (err) ppJson(err);
+    else ppJson(data);
+
+});
+----
+
+=== Setup Dynamodb on AWS
+
+Before you can use the AWS SDKs with DynamoDB, you must get an AWS access key ID and secret access key.
+For more information, see [Setting Up DynamoDB (Web Service)](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SettingUp.DynamoWebService.html).
```

Comment:
This looks a lot like a Markdown link? Is it valid in Asciidoc?

Reply:
I had a quick look at the Asciidoc reference and it doesn't look supported. I let you fix it.

Reply:
yep, fixed.